src/engine/GroupBy.cpp

```diff
@@ -916,14 +917,14 @@ GroupBy::getHashMapAggregationResults(
 // _____________________________________________________________________________
 void GroupBy::substituteGroupVariable(
     const std::vector<GroupBy::ParentAndChildIndex>& occurrences,
-    IdTable* resultTable) {
+    IdTable* resultTable) const {
   decltype(auto) groupValues = resultTable->getColumn(0);
 
   for (const auto& occurrence : occurrences) {
     sparqlExpression::VectorWithMemoryLimit<ValueId> values(
         getExecutionContext()->getAllocator());
     values.resize(groupValues.size());
-    std::copy(groupValues.begin(), groupValues.end(), values.begin());
+    std::ranges::copy(groupValues.begin(), groupValues.end(), values.begin());
```

Comment:
```suggestion
    std::ranges::copy(groupValues,  values.begin());
```
That's what sonarcloud actually wanted to say.